index.js

```diff
@@ -22,52 +24,84 @@ const fixtures = [
-    const npmResults = average(await benchmark('npm', fixture.name, {limitRuns: LIMIT_RUNS}))
-    const yarnResults = average(await benchmark('yarn', fixture.name, {limitRuns: LIMIT_RUNS}))
-    const pnpmResults = average(await benchmark('pnpm', fixture.name, {limitRuns: LIMIT_RUNS}))
-    generateSvg({
-      'npm': npmResults,
-      'yarn': yarnResults,
-      'pnpm': pnpmResults
-    }, fixture.name)
+    const npmRes = average(await benchmark('npm', fixture.name, {limitRuns: LIMIT_RUNS}))
+    const yarnRes = average(await benchmark('yarn', fixture.name, {limitRuns: LIMIT_RUNS}))
+    const pnpmRes = average(await benchmark('pnpm', fixture.name, {limitRuns: LIMIT_RUNS}))
+    const resArray = toArray(pms, {
+      'npm': npmRes,
+      'yarn': yarnRes,
+      'pnpm': pnpmRes
+    })
+
     sections.push(stripIndents`
       ${fixture.mdDesc}
 
       | action  | cache | lockfile | node_modules| npm | Yarn | pnpm |
       | ---     | ---   | ---      | ---         | --- | --- | --- |
-      | install |       |          |             | ${npmResults.firstInstall} | ${yarnResults.firstInstall} | ${pnpmResults.firstInstall} |
-      | install | ✔    | ✔        | ✔           | ${npmResults.repeatInstall} | ${yarnResults.repeatInstall} | ${pnpmResults.repeatInstall} |
-      | install | ✔    | ✔        |             | ${npmResults.withWarmCacheAndLockfile} | ${yarnResults.withWarmCacheAndLockfile} | ${pnpmResults.withWarmCacheAndLockfile} |
-      | install | ✔    |          |             | ${npmResults.withWarmCache} | ${yarnResults.withWarmCache} | ${pnpmResults.withWarmCache} |
-      | install |      | ✔        |             | ${npmResults.withLockfile} | ${yarnResults.withLockfile} | ${pnpmResults.withLockfile} |
+      | install |       |          |             | ${prettyMs(npmRes.firstInstall)} | ${prettyMs(yarnRes.firstInstall)} | ${prettyMs(pnpmRes.firstInstall)} |
+      | install | ✔    | ✔        | ✔           | ${prettyMs(npmRes.repeatInstall)} | ${prettyMs(yarnRes.repeatInstall)} | ${prettyMs(pnpmRes.repeatInstall)} |
+      | install | ✔    | ✔        |             | ${prettyMs(npmRes.withWarmCacheAndLockfile)} | ${prettyMs(yarnRes.withWarmCacheAndLockfile)} | ${prettyMs(pnpmRes.withWarmCacheAndLockfile)} |
+      | install | ✔    |          |             | ${prettyMs(npmRes.withWarmCache)} | ${prettyMs(yarnRes.withWarmCache)} | ${prettyMs(pnpmRes.withWarmCache)} |
+      | install |      | ✔        |             | ${prettyMs(npmRes.withLockfile)} | ${prettyMs(yarnRes.withLockfile)} | ${prettyMs(pnpmRes.withLockfile)} |
 
       ![Graph of the ${fixture.name} results](./results/imgs/${fixture.name}.svg)
     `)
+
+    svgs.push({
+      path: join(__dirname, 'results', 'imgs', `${fixture.name}.svg`),
+      file: generateSvg(resArray, pms)
+    })
   }
 
-  fs.writeFile('README.md', stripIndents`
-    # Node package manager benchmark
+  await Promise.all(
+    svgs.map((file) => writeFile(file.path, file.file, 'utf-8')),
```

Comment:
Does Promise.all flatten the input array?

I think it should be either

```js
await Promise.all([Promise.all(svgs.....), writeFile('README.md',...
```

or

```js
await Promise.all(svg.map(...).concat(writeFile('README.md'...
```

Reply:
There seemed to be no errors with the way I used it, but I have changed it to your first suggestion to be on the safe side